ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/ModelConformanceValidator.scala

```diff
@@ -206,9 +206,11 @@ private[transaction] class ModelConformanceValidator(engine: Engine, metrics: Me
       }
 
     val isCausallyMonotonic = transactionEntry.transaction.inputContracts.forall { contractId =>
-      val inputContractState = inputContracts(contractId)
-      val activeAt = Option(inputContractState.getActiveAt).map(parseTimestamp)
-      activeAt.exists(transactionEntry.ledgerEffectiveTime >= _)
+      // We don't check monotonicity for non-existent contracts.
```

Comment:
This is clear from the code but what's the reasoning behind the decision of letting them through here?

Reply:
I'd also appreciate some clarity here.

Reply:
My current understanding is that in case of an archived contract referenced via a `LookupByKey` we won't have the contract and the engine won't report it as `ContractNotFound` (it doesn't fetch contracts in the latter case). Hence, we'd be past model conformance and would be assuming that the contract exists, however, the lookup from `inputContracts` will fail as the contract does not exist anymore (leaving us with a `NoSuchElementException`). We don't need to check causal monotonicity for non-existent contracts hence this change (contract consistency check is at a later step).

Reply:
```suggestion
      // We don't need to check causal monotonicity now for non-existent contracts: that is part of contract consistency which is a later step
```

Reply:
@fabiotudone-da I've consolidated the previous comment and your suggestion — let me know what you think.